src/overlays/actors/ovl_Bg_Spot16_Bombstone/z_bg_spot16_bombstone.c

```diff
@@ -21,49 +110,445 @@ const ActorInit Bg_Spot16_Bombstone_InitVars = {
+
+    actor->posRot.pos.x = (sinValue * sinCosPosFactor) + actor->initPosRot.pos.x;
+    actor->posRot.pos.y = D_808B5DD8[actor->params].unk_C + actor->initPosRot.pos.y;
+    actor->posRot.pos.z = (cosValue * sinCosPosFactor) + actor->initPosRot.pos.z;
+
+    actor->shape.rot.x = D_808B5DD8[actor->params].rotation.x;
+    actor->shape.rot.y = D_808B5DD8[actor->params].rotation.y;
+    actor->shape.rot.z = D_808B5DD8[actor->params].rotation.z;
+
+    this->unk_150 = &D_060009E0;
+    this->bombiwaBankIndex = Object_GetIndex(&globalctx->objectCtx, OBJECT_BOMBIWA);
+
+    if (this->bombiwaBankIndex < 0) {
+        osSyncPrintf("Error : バンク危険！(arg_data 0x%04x)(%s %d)\n", actor->params, "../z_bg_spot16_bombstone.c",
+                     589);
+        return false;
+    }
+
+    func_808B5AF0(this);
+    return true;
+}
+
+void BgSpot16Bombstone_Init(Actor* thisx, GlobalContext* globalCtx) {
+
+    BgSpot16Bombstone* this = THIS;
+    s16 shouldLive;
+
+    func_808B4C30(this);
+
+    switch (this->actor.params) {
+        case 0xFF:
+            // The boulder is intact
+            shouldLive = func_808B4D9C(thisx, globalCtx);
+            break;
+        case 0:
+        case 1:
+        case 2:
+        case 3:
+        case 4:
+        case 5:
+            // The boulder is debris
+            shouldLive = func_808B4E58(thisx, globalCtx);
+            break;
+        default:
+            osSyncPrintf("Error : arg_data おかしいな(%s %d)(arg_data 0x%04x)\n", "../z_bg_spot16_bombstone.c", 668,
+                         this->actor.params);
+            shouldLive = false;
+            break;
+    }
+
+    if (!shouldLive) {
+        Actor_Kill(thisx);
+        return;
+    }
+    osSyncPrintf("Spot16 obj 爆弾石 (scaleX %f)(arg_data 0x%04x)\n", this->actor.scale.x, this->actor.params);
+}
+
+void BgSpot16Bombstone_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot16Bombstone* this = THIS;
+    ColliderJntSph* colliderJntSph;
+
+    if (this->actor.params == 0xFF) {
+        // Boulder is intact so remove its collider
+        colliderJntSph = &this->colliderJntSph;
+        Collider_DestroyJntSph(globalCtx, colliderJntSph);
+        Collider_DestroyCylinder(globalCtx, &this->colliderCylinder);
+    }
+}
+
+void func_808B51A8(BgSpot16Bombstone* this, GlobalContext* globalCtx) {
+    f32 scaleX1 = this->actor.scale.x * 150;
+    s16 scaleX2 = this->actor.scale.x * 250;
+    Vec3f posRot;
+
+    posRot.x = this->actor.posRot.pos.x;
+    posRot.y = this->actor.posRot.pos.y + 50.0f;
+    posRot.z = this->actor.posRot.pos.z;
+
+    func_80033480(globalCtx, &posRot, scaleX1, 2, scaleX2, 0xA0, 1);
+}
+
+void func_808B5240(BgSpot16Bombstone* this, GlobalContext* globalCtx) {
+    f32 tempUnk6;
+    f32 tempUnk2;
+    s16 index;
+    Vec3f position;
+    Vec3f* actorPosition = &this->actor.posRot.pos;
+
+    if (true) {}
+
+    while (true) {
+        if (this->unk_158 >= 26U || this->unk_154 < D_808B5EB0[this->unk_158].unk_0) {
```

Comment:
```suggestion
        if (this->unk_158 >= ARRAY_COUNT(D_808B5EB0) || this->unk_154 < D_808B5EB0[this->unk_158].unk_0) {
```

Reply:
This actually doesn't seem to match.  It looks like it's expecting the number to be unsigned and when I use the macro it becomes signed and casting it doesn't seem to help.

Reply:
there's `ARRAY_COUNTU` for unsigned, does that work

Reply:
Yeah that worked.